keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,924 @@
+  `.status.conditions` in already finished jobs.
+- Downgrade
+  - Previously configured values will be ignored, and the job will be marked
+  as completed only when all indexes succeed.
+  - the Job controller doesn't update `.status.conditions` in already finished jobs.
+
+### Version Skew Strategy
+
+The apiserver's version should be consistent with the kube-controller-manager version,
+or this feature will not work.
+
+This feature is limited to control plane.
+
+Note that, the kube-apiserver can be in the N+1 skew version relative to the
+kube-controller-manager as described [here](https://kubernetes.io/releases/version-skew-policy/#kube-controller-manager-kube-scheduler-and-cloud-controller-manager).
+In that case, the Job controller operates on the version of the Job object that
+already supports the new Job API.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: JobSuccessPolicy
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+
+No, the default behavior of a job and cronJob stays the same.
+The newly added field is optional and has to be explicitly set by the user to use this new feature.
+
+Regarding the CronJob, please see more details in [#The CronJob concurrentPolicy is not affected by JobSuccessPolicy](#the-cronjob-concurrentpolicy-is-not-affected-by-jobsuccesspolicy).
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, we can disable the `JobSuccessPolicy` feature gate.
+When the feature is disabled, the job controller stop evaluating the successPolicy even if
+the `.spec.successPolicy` is set.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The Job controller considers the `.spec.successPolicy` when it updates `.status.conditions`
+only for running Jobs and don't update `.status.conditions` for already finished jobs.
+
+###### Are there any tests for feature enablement/disablement?
+
+No, we'll add them during alpha.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Even if the kube-controller-manager is rolled out or rollback fail, already running workloads aren't any impact.
+The default behavior will be applied to running workloads.
+
+###### What specific metrics should inform a rollback?
+
+An increase in the `job_sync_duration_seconds` metrics can mean that finished jobs
+are taking longer to evaluate.
+
+The users should check whether the completed jobs have the appropriate condition,
+specifically the reason.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No.
```

Comment:
Should be done before promoting to beta - if you're filling it, let's switch to "TODO" or sth (or leave it empty for Alpha - same for other questions not necessary required now).

Reply:
You're right. I will replace `No` with `TODO`.

Reply:
Done.